src/providers/oracle/qgsoracleprovider.cpp

```diff
@@ -1301,11 +1301,14 @@ bool QgsOracleProvider::addFeatures( QgsFeatureList &flist, QgsFeatureSink::Flag
         QVariant value = attributevec.value( fieldId[i], QVariant() );
 
         QString v;
-        if ( !value.isValid() )
+        if ( !value.isValid() || value.isNull() )
```

Comment:
I think `isNull()` also returns true for invalid QVariants, so the first test could be skipped.